### DISCUSSION TITLE
`latex-...` options not picked up?

### DISCUSSION THREAD

**Discussion by devmotion** [quarto-dev/quarto-cli#4734] [Q&A]
### Bug description

Hej!

In a larger project, I would like to use my own `latexmk` pipeline because manually running `latexmk` with the LaTeX file generated by Quarto works successfully whereas compilation is unsuccessful when I try to generate the PDF with quarto and `pdf-engine: latexmk`. Based on the docs, I assumed I could opt out of Quarto's machinery by setting `latex-auto-mk: false`. However, changing this setting does seem to make a difference (as someone else had noticed before apparently: https://github.com/quarto-dev/quarto-cli/discussions/819#discussioncomment-2704475). In contrast to other configurations, these options also don't show up in the pandoc/metadata output in the terminal.

So I tried to set some of the other `latex-...` options and they also did not seem to affect compilation. For instance, I played around with the following somewhat minimal example:
```md
---
title: MWE
format:
  pdf:
    pdf-engine: latexmk
    pdf-engine-opt: '-verbose'
    latex-auto-mk: false
    latex-output-dir: "/home/david/quarto-mwe/"
---

A MWE for debugging the `latex-` options.
```
I rendered it by running
```shell
quarto render mwe.qmd --log-level=debug
```
in the terminal. Both with Quarto 1.2.335 and 1.3.34 (Fedora 37), the `latex-...` options do not change the output in the terminal and, e.g., regardless of the option `latex-output-dir` always a directory `/tmp/...` is used by latexmk.

Am I doing something wrong or do these options not work currently?

### Checklist

- [ ] Please include a minimal, fully reproducible example in a single .qmd file? Please provide the whole file rather than the snippet you believe is causing the issue.
- [ ] Please [format your issue](https://quarto.org/bug-reports.html#formatting-make-githubs-markdown-work-for-us) so it is easier for us to read the bug report.
- [ ] Please document the RStudio IDE version you're running (if applicable), by providing the value displayed in the "About RStudio" main menu dialog?
- [ ] Please document the operating system you're running. If on Linux, please provide the specific distribution.

**Comment by cderv** [comment]
@dragonstyle is ` latex-output-dir` config supposed to be working only with Quarto latexmk or not ? It seems from our code it is is, but not sure if it is an oversight or not. Thanks

**Reply by devmotion** [reply]
To be clear, for me `latex-output-dir` doesn't make a difference regardless of whether I use `latex-auto-mk: true` or `latex-auto-mk: false`.

**Reply by dragonstyle** [reply]
I believe that this is only an option for configuring the Quarto latex engine. If you're using the latexmk engine directly, you should use the engine options to configure it as you'd like.

**Reply by devmotion** [reply]
Thanks for this clarification! 

This does not explain the following behaviour though, does it?
- `latex-output-dir` is not respected if `latex-auto-mk: true` (and neither if `latex-auto-mk: false`)
- a `.latexmkrc` file with `$out_dir = "...";` is neither respected if `latex-auto-mk: true` nor if `latex-auto-mk: false`

So to summarize, to me it seems by setting `pdf-engine: latexmk` quarto always runs a default pipeline for latexmk, regardless of which options I set in quarto or in a `.latexmkrc` config file. Only `pdf-engine-opt` seems to be respected, this allows me e.g. to set `-output-directory=...`.

**Reply by cderv** [reply]
I wonder if we could have some specific handling that would miss the `.latexmkrc` config... 

We have `pdf-engine-opt` config that you could set though, like in `pdf-engine-opt: -outdir=out` shown in https://quarto.org/docs/output-formats/pdf-engine.html#alternate-pdf-engines

Is this at least working ?

**Reply by devmotion** [reply]
Yes, as mentioned above `pdf-engine-opt` is respected and I can set e.g. the output directory with `-output-directory=...`. But all `latex-...` options do not seem to work - even if I use the Quarto pipeline with `latex-auto-mk: true` (which is the default anyway, as far as I understand). And it seems also `latex-auto-mk: false` does not have any effect.

It seems someone else made the same observation https://github.com/quarto-dev/quarto-cli/discussions/819#discussioncomment-2704475, so I think it should be possible to reproduce this behaviour.
